docs/source/user_guide/automl.ipynb

```diff
@@ -71,6 +71,35 @@
     "The AutoML interface supports a variety of other parameters. For a comprehensive list, please [refer to the API reference.](../generated/evalml.automl.AutoMLSearch.ipynb)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Detecting Problem Type\n",
+    "\n",
+    "EvalML includes a simple method, `detect_problem_type`, to help determine the problem type given the target data. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from evalml.problem_types import detect_problem_type\n",
+    "\n",
+    "y = pd.Series([0, 1, 1, 0, 1, 1])\n",
+    "detect_problem_type(y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This function can return the predicted problem type as a ProblemType enum, choosing from ProblemType.BINARY, ProblemType.MULTICLASS, and ProblemType.REGRESSION. If the target data is invalid (for instance when there is only 1 unique label), the function will throw an error instead."
```

Comment:
I suggest you move this up to the first sentence, before the code example, and make some minor tweaks to the text to simplify: So in total:
> EvalML includes a simple method `detect_problem_type` which returns the machine learning problem type deemed most useful for modeling the specified target.

I like what you said about the return types and the error! I think that's great stuff for the API docs. The user guide is intended as a quick high-level dive into using evalml.